hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -385,25 +409,27 @@ private static void modify(FileSystem fs, Path metadataFolder, Properties modify
       // 0. do any recovery from prior attempts.
       recoverIfNeeded(fs, cfgPath, backupCfgPath);
 
-      // 1. backup the existing properties.
-      try (FSDataInputStream in = fs.open(cfgPath);
-           FSDataOutputStream out = fs.create(backupCfgPath, false)) {
-        FileIOUtils.copy(in, out);
+      // 1. Read the existing config
+      TypedProperties props = fetchConfigs(fs, metadataFolder.toString());
+
+      // 2. backup the existing properties.
+      try (FSDataOutputStream out = fs.create(backupCfgPath, false)) {
+        storeProperties(props, out);
       }
```

Comment:
don't think we can easily handle concurrent update scenario without a proper lock, given FS like cloud store does not support atomic write. When multi-writer is used (check the config), can we incorporate lock manager into this API if applicable?

Reply:
For the hoodie.properties updates caused by MDT in multi-writer there should already be locking as all MDT operations are performed under the dataset lock already. For multi-writer scenarios locking is already required.

In non-MDT cases, I don't see any properties that are updated generally by multiple-writers. So the risk of this writer-writer conflict is very low.

Reply:
Best handled in a separate ticket as we will need to identify all codepaths and adding locking there wont be trivial.